contracts/extensions/erc1155/ERC1155TokenExtension.sol

```diff
@@ -0,0 +1,361 @@
+
+    // The internal owner of record of an NFT that has been transferred to the extension
+    mapping(bytes32 => address) private _ownership;
+
+    //the number amount of Token IDs that belong to an NFT address stored in the Guild for a specific Token ID
+    // owner => tokenAddress => tokenId => tokenAmount
+    mapping(address => mapping(address => mapping(uint256 => EnumerableSet.UintSet)))
+        private _nftTracker;
+
+    //All the NFT addresses collected and stored in the GUILD collection
+    EnumerableSet.AddressSet private _nftAddresses;
+
+    //MODIFIERS
+    modifier hasExtensionAccess(IExtension extension, AclFlag flag) {
+        require(
+            dao.state() == DaoRegistry.DaoState.CREATION ||
+                dao.hasAdapterAccessToExtension(
+                    msg.sender,
+                    address(extension),
+                    uint8(flag)
+                ),
+            "1155TokenExtension::accessDenied"
+        );
+        _;
+    }
+
+    /// @notice Clonable contract must have an empty constructor
+    constructor() {}
+
+    /**
+     * @notice Initializes the extension with the DAO address that it belongs to.
+     * @param _dao The address of the DAO that owns the extension.
+     * @param creator The owner of the DAO and Extension that is also a member of the DAO.
+     */
+    function initialize(DaoRegistry _dao, address creator) external override {
+        require(!initialized, "already initialized");
+        require(_dao.isMember(creator), "not a member");
+
+        initialized = true;
+        dao = _dao;
+    }
+
+    /**
+     * @notice Collects the NFT from the owner and moves it to the NFT extension.
+     * @notice It must be have been allowed to move this token by either {approve} or {setApprovalForAll}.
+     * @dev Reverts if the NFT is not in ERC1155 standard.
+     * @param nftAddr The NFT contract address.
+     * @param nftTokenId The NFT token id.
+     * @param amount The amount of NFT with nftTokenId to be collected.
+     */
+    function collect(
+        address nftAddr,
+        uint256 nftTokenId,
+        uint256 amount
+    ) external hasExtensionAccess(this, AclFlag.COLLECT_NFT) {
+        IERC1155 erc1155 = IERC1155(nftAddr);
+        // check balance of the NFT to the contract address - unlike 721 no 'ownerOf'
+        uint256 ownerTokenIdBalance =
+            erc1155.balanceOf(address(this), nftTokenId);
+
+        //If the NFT is already in the NFTExtension, update the ownership if not set already
+        if (ownerTokenIdBalance > 0) {
+            if (_ownership[getNFTId(nftAddr, nftTokenId)] == address(0x0)) {
+                _saveNft(nftAddr, nftTokenId, GUILD, amount);
+
+                emit CollectedNFT(nftAddr, nftTokenId, amount);
+            }
+            //If the NFT is not in the NFTExtension, we try to transfer from the current owner of the NFT to the extension
+        } else {
+            erc1155.safeTransferFrom(
+                msg.sender,
```

Comment:
The `msg.sender` at this state is actually the adapter address, so it considers the adapter already holds the asset to be able to transfer it. If it is not the case, we might need to pass the actual owner address in the arguments.

Reply:
Yeah, so we need to make this function open (without any ACL).